code/modules/antagonists/traitor/contractor/datums/syndicate_contract.dm

```diff
@@ -0,0 +1,518 @@
+  * * target_dead - Whether the target was extracted dead.
+  */
+/datum/syndicate_contract/proc/notify_completion(tc = 0, creds = 0, target_dead = FALSE)
+	var/penalty_text = ""
+	if(target_dead)
+		penalty_text = " (penalty applied as the target was extracted dead)"
+	owning_hub.uplink?.message_holder("Well done, agent. The package has been received and will be processed shortly before being returned. "\
+									 + "As agreed, you have been credited with [tc] telecrystals[penalty_text] and [creds] credits.", 'sound/machines/terminal_prompt_confirm.ogg')
+
+/**
+  * Handles the target's experience from extraction.
+  *
+  * Arguments:
+  * * M - The target mob.
+  * * P - The extraction portal.
+  */
+/datum/syndicate_contract/proc/handle_target_experience(mob/M, obj/effect/portal/redspace/contractor/P)
+	var/turf/T = get_turf(P)
+	var/mob/living/carbon/human/H = M
+
+	// Prepare their return
+	addtimer(CALLBACK(src, .proc/handle_target_return, M, T), prison_time)
+
+	// Shove all of the victim's items in the secure locker.
+	victim_belongings = list()
+	var/list/obj/item/stuff_to_transfer = list()
+	for(var/obj/item/I in M)
+		// Any items we don't want to take from them?
+		if(istype(H))
+			// Keep their uniform and shoes
+			if(I == H.w_uniform || I == H.shoes)
+				continue
+			// Plasmamen are no use if they're crispy
+			if(isplasmaman(H) && I == H.head)
+				continue
+
+		// Any kind of implant gets removed (mindshield, etc)
+		if(istype(I, /obj/item/implant))
+			qdel(I)
+			continue
+
+		if(M.unEquip(I))
+			stuff_to_transfer += I
+
+	// Cybernetic implants as well
+	for(var/obj/item/organ/internal/cyberimp/I in H.internal_organs)
+		// Greys get to keep their implant
+		if(isgrey(H) && istype(I, /obj/item/organ/internal/cyberimp/brain/speech_translator))
+			continue
+		// Try removing it
+		I = I.remove(H)
+		if(I)
+			stuff_to_transfer += I
+
+	// Transfer it all (or drop it if not possible)
+	for(var/i in stuff_to_transfer)
+		var/obj/item/I = i
+		if(GLOB.prisoner_belongings.give_item(I))
+			victim_belongings += I
+		else if(!((ABSTRACT|NODROP) in I.flags)) // Anything that can't be put on hold, just drop it on the ground
+			I.forceMove(T)
+
+	// Give some species the necessary to survive. Courtesy of the Syndicate.
+	if(istype(H))
+		var/obj/item/tank/emergency_oxygen/tank
+		var/obj/item/clothing/mask/breath/mask
+		if(isvox(H))
+			tank = new /obj/item/tank/emergency_oxygen/nitrogen(H)
+			mask = new /obj/item/clothing/mask/breath/vox(H)
+		else if(isplasmaman(H))
+			tank = new /obj/item/tank/emergency_oxygen/plasma(H)
+			mask = new /obj/item/clothing/mask/breath(H)
+
+		if(tank)
+			H.equip_to_appropriate_slot(tank)
+			H.equip_to_appropriate_slot(mask)
+			tank.toggle_internals(H, TRUE)
+
+	M.update_icons()
+
+	// Narrate their kidnapping and torturing experience.
+	if (M.stat != DEAD)
+		// Heal them up - gets them out of crit/soft crit.
+		M.reagents.add_reagent("omnizine", 20)
```

Comment:
Maybe give a bit of food as well? (Though maybe better to put it in a cell)

Reply:
The food might be a bit moldy.